diff --git a/src/contrastors/layers/attention.py b/src/contrastors/layers/attention.py
index bdd28b7..448a6dc 100644
--- a/src/contrastors/layers/attention.py
+++ b/src/contrastors/layers/attention.py
@@ -153,7 +153,7 @@ class FlashAttention(nn.Module):
             if self.num_heads == self.num_heads_kv:
                 if cu_seqlens is None and max_seq_len is None:
                     bsz, q_len, h_size = hidden_states.shape
-                    unpadded_qkv, indices, cu_seqlens, max_seq_len = unpad_input(qkv, attention_mask)
+                    unpadded_qkv, indices, cu_seqlens, max_seq_len, *_ = unpad_input(qkv, attention_mask)
 
                     attn_outputs = flash_attn_varlen_qkvpacked_func(
                         unpadded_qkv,
@@ -183,8 +183,8 @@ class FlashAttention(nn.Module):
             else:
                 if cu_seqlens is None and max_seq_len is None:
                     bsz, q_len, h_size = hidden_states.shape
-                    unpadded_q, indices_q, cu_seqlens, max_seq_len = unpad_input(q, attention_mask)
-                    unpadded_kv, _, cu_seqlens_k, max_seq_len_k = unpad_input(kv, attention_mask)
+                    unpadded_q, indices_q, cu_seqlens, max_seq_len, *_ = unpad_input(q, attention_mask)
+                    unpadded_kv, _, cu_seqlens_k, max_seq_len_k, *_ = unpad_input(kv, attention_mask)
 
                     attn_outputs = flash_attn_varlen_kvpacked_func(
                         q=unpadded_q,
@@ -385,9 +385,9 @@ class FlashAttentionPooling(nn.Module):
         if attention_mask is not None:
             # varlen, ignore padding tokens, efficient for large batch with many paddings
             assert attention_mask is not None
-            unpadded_q, _, cu_seqlens, max_seqlen = unpad_input(q, torch.ones_like(attention_mask))
+            unpadded_q, _, cu_seqlens, max_seqlen, *_ = unpad_input(q, torch.ones_like(attention_mask))
             if cu_seqlens_k is None and max_seqlen_k is None:
-                unpadded_kv, indices_kv, cu_seqlens_k, max_seqlen_k = unpad_input(kv, attention_mask)
+                unpadded_kv, indices_kv, cu_seqlens_k, max_seqlen_k, *_ = unpad_input(kv, attention_mask)
 
                 attn_outputs = flash_attn_varlen_kvpacked_func(
                     unpadded_q,
diff --git a/src/contrastors/layers/moe.py b/src/contrastors/layers/moe.py
index df65f19..20f11b1 100644
--- a/src/contrastors/layers/moe.py
+++ b/src/contrastors/layers/moe.py
@@ -71,7 +71,7 @@ class MoEBlock(nn.Module):
         input_shape = hidden_states.shape
         batch, seqlen = hidden_states.shape[:2]
         if attention_mask is not None:
-            hidden_states, indices, _, _ = unpad_input(hidden_states, attention_mask)
+            hidden_states, indices, _, _, *_ = unpad_input(hidden_states, attention_mask)
         hidden_states = hidden_states.reshape(-1, input_shape[-1])
         # (bs * s, num_experts)
         # TODO: could use fp32 here as switch transformer found it didn't diverge
diff --git a/src/contrastors/models/biencoder/flash_llama.py b/src/contrastors/models/biencoder/flash_llama.py
index a0cba20..8225233 100644
--- a/src/contrastors/models/biencoder/flash_llama.py
+++ b/src/contrastors/models/biencoder/flash_llama.py
@@ -400,8 +400,8 @@ class LlamaAttention(nn.Module):
 
             assert attention_mask is not None
 
-            unpadded_kv, indices_k, cu_seqlens_k, max_seqlen_k = unpad_input(kv, attention_mask)
-            unpadded_q, indices_q, cu_seqlens_q, max_seqlen_q = unpad_input(q, attention_mask)
+            unpadded_kv, indices_k, cu_seqlens_k, max_seqlen_k, *_ = unpad_input(kv, attention_mask)
+            unpadded_q, indices_q, cu_seqlens_q, max_seqlen_q, *_ = unpad_input(q, attention_mask)
             attn_outputs = flash_attn_varlen_kvpacked_func(
                 unpadded_q,
                 unpadded_kv,
diff --git a/src/contrastors/models/biencoder/flash_pythia.py b/src/contrastors/models/biencoder/flash_pythia.py
index 6ae2dfd..0492e7a 100644
--- a/src/contrastors/models/biencoder/flash_pythia.py
+++ b/src/contrastors/models/biencoder/flash_pythia.py
@@ -240,8 +240,8 @@ class GPTNeoXAttention(nn.Module):
 
             assert attention_mask is not None
 
-            unpadded_kv, indices_k, cu_seqlens_k, max_seqlen_k = unpad_input(kv, attention_mask)
-            unpadded_q, indices_q, cu_seqlens_q, max_seqlen_q = unpad_input(query, attention_mask)
+            unpadded_kv, indices_k, cu_seqlens_k, max_seqlen_k, *_ = unpad_input(kv, attention_mask)
+            unpadded_q, indices_q, cu_seqlens_q, max_seqlen_q, *_ = unpad_input(query, attention_mask)
             attn_outputs = flash_attn_varlen_kvpacked_func(
                 unpadded_q,
                 unpadded_kv,
diff --git a/src/contrastors/models/biencoder/modeling_biencoder.py b/src/contrastors/models/biencoder/modeling_biencoder.py
index a72e76a..3872151 100644
--- a/src/contrastors/models/biencoder/modeling_biencoder.py
+++ b/src/contrastors/models/biencoder/modeling_biencoder.py
@@ -130,7 +130,7 @@ class MultiHeadAttentionPooling(nn.Module):
 
     def forward(self, hidden_states, input_ids, attention_mask):
         if attention_mask is not None:
-            hidden_states, indices, cu_seqlens, max_seqlen_in_batch = unpad_input(hidden_states, attention_mask)
+            hidden_states, indices, cu_seqlens, max_seqlen_in_batch, *_ = unpad_input(hidden_states, attention_mask)
         else:
             indices = None
             cu_seqlens = None
diff --git a/src/contrastors/models/encoder/modeling_nomic_bert.py b/src/contrastors/models/encoder/modeling_nomic_bert.py
index ba43836..2973f3f 100644
--- a/src/contrastors/models/encoder/modeling_nomic_bert.py
+++ b/src/contrastors/models/encoder/modeling_nomic_bert.py
@@ -330,7 +330,9 @@ class NomicBertEncoder(NomicBertPreTrainedModel):
 
         batch, seqlen = hidden_states.shape[:2]
         if not self.gradient_checkpointing and getattr(self.config, "moe_every_n_layers", 0) <= 0:
-            hidden_states, indices, cu_seqlens, max_seqlen_in_batch = unpad_input(hidden_states, attention_mask)
+            hidden_states, indices, cu_seqlens, max_seqlen_in_batch, *_ = unpad_input(
+                hidden_states, attention_mask
+            )
         else:
             indices, cu_seqlens, max_seqlen_in_batch = None, None, None
 
